src/integrations/prefect-dbt/prefect_dbt/core/task_state.py

```diff
@@ -0,0 +1,133 @@
+"""
+State for managing tasks across callbacks.
+"""
+
+import threading
+from typing import Any, Union
+
+from prefect.client.schemas.objects import State
+from prefect.context import hydrated_context
+from prefect.task_engine import run_task_sync
+from prefect.tasks import Task
+
+
+class TaskState:
```

Comment:
this should be better named and private (i.e. prefixed by an underscore) or instead placed in a `_private_module`